contrib/babelfishpg_tsql/runtime/functions.c

```diff
@@ -2088,6 +2089,292 @@ numeric_radians(PG_FUNCTION_ARGS)
 	PG_RETURN_NUMERIC(result);
 }
 
+static inline int32_t
+GetUTF8CodePoint(const unsigned char *in, int len, int *consumed_p)
```

Comment:
Wondering why we need to worry about encodings here. The design doc also mentioned needing to calculate length in bytes under UTF-16. Is this the first time we've needed a function to get a UTF-8 code point?